src/stateMachineFwk/controlBlock.cpp

```diff
@@ -149,31 +165,53 @@ namespace SM
 		pdb_mp = pdb_p;
 	}
 
-    TempDataBlock* ControlBlock::getTempDataBlock() const
+    void ControlBlock::addTempDataBlock(TempDataBlock* tdb_p)
     {
-    	return tdb_mp;
+        if (tdb_p != NULL)
+        {
+        	tdb_p->setNextTempDataBlock(tdb_mp);
```

Comment:
this seems to be adding temp data lock at the head...why ?

Reply:
We are supporting multiple procedures at the same time. New procedures allocated will get added to the head of the queue.